### PR TITLE
gluon-mmfd: add package

### PR DESCRIPTION
**Pull request by christf**
This adds the gluon integration needed for mmfd when using the socket features avoiding restarts of mmfd on every (attempted) mesh interface change.

### CODE REVIEW

File: package/gluon-mmfd/files/etc/init.d/mmfd [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	do
+		if ! mmfd_has_interface $i
+		then
+			addif $i
+		fi
+	done
+
+	for i in $(mmfd_get_interfaces)
+	do
+		if ! ubus call network.interface dump | jsonfilter -e "@.interface[@.proto='gluon_mesh' && @.up=true].device"|grep -q "^$i$"
```
Comment (CodeFetch) [inline package/gluon-mmfd/files/etc/init.d/mmfd]: add spaces before and after bar before grep due to consistency.

File: package/gluon-mmfd/Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gluon-mmfd
+PKG_VERSION:=1
+
+include ../gluon.mk
+
+define Package/gluon-mmfd
+  TITLE:=mesh multicast forwarding daemon - gluon integration
```
Comment (rotanid) [inline package/gluon-mmfd/Makefile]: maybe one or another capital letter would be nice here, especially for "Gluon", maybe also "Mesh" as it's the beginning

File: package/gluon-mmfd/files/etc/init.d/mmfd [hunk 3]
```diff
@@ -0,0 +1,64 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=50
+
+DAEMON=/usr/sbin/mmfd
+SOCKET=/var/run/mmfd.sock
+
+start_service() {
+	local interfaces=$(for dev in $(gluon-list-mesh-interfaces); do echo " -i $dev";done)
```
Comment (rotanid) [inline package/gluon-mmfd/files/etc/init.d/mmfd]: very small thing, but a space before "done" ?

File: package/gluon-mmfd/files/etc/init.d/mmfd [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+service_triggers() {
+	local script=$(readlink "$initscript")
+	local name=$(basename "${script:-$initscript}")
+
+	procd_open_trigger
+	procd_add_raw_trigger "interface.*" 0 "/etc/init.d/$name" reload
+	procd_close_trigger
+}
+
```
Comment (rotanid) [inline package/gluon-mmfd/files/etc/init.d/mmfd]: empty line at the end

Comment (christf) [reply package/gluon-mmfd/files/etc/init.d/mmfd]: Das ist Platz für Notizen! :-)

File: package/gluon-mmfd/Makefile (hunk 2 above)
Comment (christf) [reply package/gluon-mmfd/Makefile]: ok.

File: package/gluon-mmfd/files/etc/init.d/mmfd (hunk 3 above)
Comment (christf) [reply package/gluon-mmfd/files/etc/init.d/mmfd]: (y)

**Comment by rotanid** [comment]
i merged it now because of the small size and no impact on existing setups

**Pull request merged**